diff -Naur nx-X11/config/imake/imake.c nx-X11-fix-format-errors/config/imake/imake.c
--- nx-X11/config/imake/imake.c	2005-11-08 07:33:24.000000000 +0100
+++ nx-X11-fix-format-errors/config/imake/imake.c	2009-03-04 22:25:31.000000000 +0100
@@ -1015,7 +1015,7 @@
     abort ();
 
   while (fgets (command, len, fp))
-    fprintf (inFile, command);
+    fprintf (inFile, "%s", command);
 
   len = pclose (fp);
   remove (aout);
